plugin/api.go

```diff
@@ -113,6 +113,12 @@ type API interface {
 
 	// Delete will remove a key-value pair. Returns nil for non-existent keys.
 	KVDelete(key string) *model.AppError
+
+	// PublishWebSocketEvent sends an event to WebSocket connections.
+	// event is the type and will be prepended with "custom_<pluginid>_"
+	// payload is the data sent with the event. Interface values must be primitive Go types or mattermost-server/model types
+	// broadcast determines which users to send the event too
+	PublishWebSocketEvent(event string, payload map[string]interface{}, broadcast *model.WebsocketBroadcast)
```

Comment:
`payload` is a `map[string]interface{}` for two reasons:

1. To send custom struct types across RPC they need to registered on both RPC clients (the server-side and the plugin-side), which isn't possible as the server won't know about any structs the plugin creates
2. Our WebsocketEvent struct's `Data` field is already `map[string]interface{}` and it would be a relatively large amount of work to switch that over to just be an `interface{}`